src/Hamiltonians/TwoBodyCorrelation.jl

```diff
@@ -0,0 +1,74 @@
+# Methods that need to be implemented:
+#   •  starting_address(::AbstractHamiltonian) - not needed
+"""
+    G2Correlator(d::Int)
+
+Implements the two-body correlation operator for computing the correlation
+between two particles in two different components of a Fock states encoded
+in momentum space. The variable `d` is the distance between the two particles
+which can take `Integer` values from `0` to `M` (the number of modes).
+It returns a `Complex` value.
+It currently only works on [`BoseFS2C`](@ref).
+
+
+```math
+\\hat{G}_2^{(d)} = \\frac{1}{M}\\sum_{spqr=1}^M e^{-id(p-q)2π/M} a^†_{s} b^†_{p}  b_q a_r δ_{s+p,q+r}
```

Comment:
```suggestion
\\hat{G}_2(d) = \\frac{1}{M}\\sum_{spqr=1}^M e^{-id(p-q)2π/M} a^†_{s} b^†_{p}  b_q a_r δ_{s+p,q+r}
```
This is how I would write it. G_2 is a function of `d`.